src/j1client.test.js

```diff
@@ -0,0 +1,68 @@
+const { Polly } = require("@pollyjs/core");
+const NodeHTTPAdapter = require("@pollyjs/adapter-node-http");
+const JupiterOneClient = require("./j1client");
+
+Polly.register(NodeHTTPAdapter);
+
+const FAKE_ACCOUNT = "johndoe";
+const FAKE_KEY = "abc123";
+
+const j1qlString = "Find jupiterone_account";
+const mockResponse = {
+  data: {
+    queryV1: {
+      data: [],
+      __typename: "QueryV1Response"
+    }
+  }
+};
+
+let polly;
+let j1Client;
+let attempts = 0;
+let attemptsToFail = 0;
+
+beforeEach(async () => {
+  attempts = 0;
+
+  j1Client = await new JupiterOneClient({
+    account: FAKE_ACCOUNT,
+    accessToken: FAKE_KEY
+  }).init();
+
+  polly = new Polly("JupiterOneClient tests", {
+    adapters: ["node-http"]
+  });
+  polly.server.any().intercept((req, res) => {
+    attempts++;
+    res.status(attempts <= attemptsToFail ? 401 : 200);
+    res.json(mockResponse);
+  });
+});
+
+afterEach(() => {
+  return polly.stop(); // returns a Promise
+});
+
+describe("failing 4 times", () => {
+  beforeEach(() => {
+    attemptsToFail = 4;
+  });
+
+  test("client retries failed requests and returns successful response", async () => {
+    const results = await j1Client.queryV1(j1qlString);
+    expect(results.length).toBe(0);
+  }, 10000);
+});
+
+describe("failing 5 times", () => {
+  beforeEach(() => {
+    attemptsToFail = 5;
+  });
+
+  test("client retries 5 times and throws error", async () => {
+    await expect(j1Client.queryV1(j1qlString)).rejects.toThrow(
+      /Network error: Response not successful: Received status code 401/
+    );
+  }, 10000);
```

Comment:
I believe the timeout argument for `test` can be dropped since everything runs super fast with the polly server.

Reply:
Unfortunately it can't, since the retrying solution involves exponential backoff. This test case usually takes between 5 and 7 seconds.

Reply:
It also jitters so I don't know precisely how long it will take 😄

Reply:
Ahhh, gotcha. Thanks for explaining!

I think this will be as good as it gets with our API today, but eventually it would be good for our API to change to include headers that inform the client of rate limit status and how long to wait before retry, so that clients can run as fast as possible. Some day!